func.go

```diff
@@ -89,37 +116,41 @@ func (e *FuncExec) CommandContext(ctx context.Context, name string, arg ...strin
 	return cmd
 }
 
-// findFunc retrives a function and the command name from the func map
-func (e *FuncExec) findFunc(name string) (CmdFunc, string) {
-	// dont even need to check the map if it's nil
-	if e.funcMap == nil {
-		return nil, ""
-	}
+// FuncExecOption can be used to configure the FuncExec struct
+type FuncExecOption func(*FuncExec)
 
-	// check if it's a simple member of the map
-	if fn, ok := e.funcMap[name]; ok {
-		return fn, name
-	}
+// WithFuncMap adds new handlers for each of the functions in the func map.
+// The keys are used to match commands by their names.
+// This also adds the keys to the pseudo $PATH used by exec
+func WithFuncMap(funcs map[string]CmdFunc) FuncExecOption {
+	return func(fExec *FuncExec) {
+		if fExec.mux == nil {
+			fExec.mux = NewMux()
+		}
 
-	// check if there's a path that matches
-	// e.g. go -> /usr/local/go/bin/go
-	for file, fn := range e.funcMap {
-		if filepath.Base(file) == name {
-			return fn, file
+		var addLast CmdFunc
+		for key, fn := range funcs {
+			fExec.bins = append(fExec.bins, key)
+
+			// make sure the catch-all case is the final matcher
+			if key == "*" {
+				addLast = fn
+				continue
+			}
+
+			fExec.mux.HandleFunc(pat(key), fn)
 		}
-	}
 
-	// no match was found
-	return nil, ""
+		if addLast != nil {
+			fExec.mux.HandleFunc(pat("*"), addLast)
+		}
+	}
 }
 
-// FuncExecOption can be used to configure the FuncExec struct
-type FuncExecOption func(*FuncExec)
-
-// WithFuncMap sets the func map used by all the commands created by this Exec
-func WithFuncMap(funcs map[string]CmdFunc) FuncExecOption {
+// WithBins adds the specified bins to the pseudo $PATH used by this Exec
+func WithBins(bins ...string) FuncExecOption {
 	return func(fExec *FuncExec) {
-		fExec.funcMap = funcs
+		fExec.bins = append(fExec.bins, bins...)
```

Comment:
Should this still be a functional option or should it be a variadic to the NewFuncExec function?